client/index.js

```diff
@@ -42,7 +43,7 @@ window.__fcc__ = {};
 
 const serviceOptions = { xhrPath: '/services', context: { _csrf: csrfToken } };
 
-const history = useLangRoutes(createHistory)();
+const history = useLangRoutes(createHistory, primary)();
```

Comment:
This state is already in the url. 

Can we add a closure to useLangRoutes and pull this information from the url instead of relying on the initialState?